Index: trunk/src/Makefile.mingw
===================================================================
--- trunk/src/Makefile.mingw	(revision 3649)
+++ trunk/src/Makefile.mingw	(working copy)
@@ -24,6 +24,9 @@
 XTLIB =  
 GTKINC = `gtk-config --cflags`
 GTKLIB = `gtk-config --libs`
+HAVE_GLIB = $(shell pkg-config --exists glib-2.0 && echo 'yes')
+GLIBINC = `pkg-config --cflags glib-2.0`
+GLIBLIB = `pkg-config --libs glib-2.0`
 GLUTINC = -I/usr/include -I.
 GLUTLIB = -L.
 TCLINCL	= -I/usr/include/tcl8.4
@@ -48,7 +51,7 @@
 
 TOOLS_DIR = ../tools
 CC=gcc
-CFLAGS =  -DWIN32 -DWINVER=0x0601 -D_WIN32_WINNT=0x0601  $(REGEXP) $(PCREINC)
+CFLAGS =  -DWIN32 -DWINVER=0x0601 -D_WIN32_WINNT=0x0601  $(REGEXP) $(PCREINC) $(GLIBINC)
 # IVY full debug
 #CFLAGS = -g  -DDEBUG -DWIN32 -DWINVER=0x0601 -D_WIN32_WINNT=0x0601  $(REGEXP) $(PCREINC)
 OBJ = ivyloop.o timer.o ivysocket.o ivy.o ivybuffer.o ivyfifo.o ivybind.o intervalRegexp.o param.o
@@ -65,9 +68,14 @@
 .c.o:
 	$(CC) $(CFLAGS) -c $*.c
 
-all: static-libs  shared-libs tools
+all: static-libs  shared-libs tools pkgconf
 
+ifeq ($(HAVE_GLIB),yes)
+static-libs: libivy.a libgivy.a libglibivy.a libtclivy.a
+else
 static-libs: libivy.a libgivy.a
+endif
+#static-libs: libivy.a libgivy.a
 #static-libs: libivy.a libgivy.a libtclivy.a  
 # not yiet libxtivy.a libgtkivy.a  need Modified Glut libglutivy.a
 
@@ -87,6 +95,9 @@
 ivygtkloop.o: ivygtkloop.c ivygtkloop.h
 	$(CC) -c $(CFLAGS) $(GTKINC) -o ivygtkloop.o ivygtkloop.c
 
+ivyglibloop.o: ivyglibloop.c ivyglibloop.h
+	$(CC) -c $(CFLAGS) $(GLIBINC) -o ivyglibloop.o ivyglibloop.c
+
 ivyprobe: ivyprobe.o libivy.a
 	$(CC)  $(CFLAGS) -o ivyprobe ivyprobe.o -L. -livy -lwsock32 $(PCRELIB) 
 
@@ -124,6 +135,10 @@
 	$(RM) $@
 	ar cq $@ $(GTKOBJ)
 
+libglibivy.a: $(GLIBOBJ)
+	$(RM) $@
+	ar cq $@ $(GLIBOBJ)
+
 libglutivy.a: $(GLUTOBJ)
 	$(RM) $@
 	ar cq $@ $(GLUTOBJ)
@@ -208,6 +223,11 @@
 tools: static-libs
 	@(cd $(TOOLS_DIR) && $(MAKE) -f Makefile.mingw)
 
+pkgconf:
+	for f in *.pc.in ; do \
+		sed -e 's,@PREFIX@,$(PREFIX),; s,@MAJOR@,$(MAJOR),; s,@MINOR@,$(MINOR),; s,@PCRELIB@,$(PCRELIB),; s,@EXTRALIB@,$(EXTRALIB),' $$f > $$(echo $$f | cut -f 1,2 -d .); \
+	done
+
 rpm::
 	/usr/bin/rpmize
 
Index: trunk/src/ivy.c
===================================================================
--- trunk/src/ivy.c	(revision 3649)
+++ trunk/src/ivy.c	(working copy)
@@ -29,7 +29,7 @@
 #include <windows.h>
 #include "timer.h"
 #define snprintf _snprintf
-#ifdef __MINGW32__
+#if !defined(__MINGW32__)
 // should be removed in when defined in MinGW include of ws2tcpip.h
 extern const char * WSAAPI inet_ntop(int af, const void *src,
                              char *dst, socklen_t size);
@@ -69,6 +69,18 @@
 #define DEFAULT_DOMAIN 127.255.255.255
 #endif
 
+#ifndef timersub
+# define timersub(a, b, result)						\
+	do {								\
+		(result)->tv_sec = (a)->tv_sec - (b)->tv_sec;		\
+		(result)->tv_usec = (a)->tv_usec - (b)->tv_usec;        \
+		if ((result)->tv_usec < 0) {                            \
+			--(result)->tv_sec;				\
+			(result)->tv_usec += 1000000;			\
+		}                                                       \
+	} while (0)
+#endif
+
 /* stringification et concatenation du domaine et du port en 2 temps :
  * Obligatoire puisque la substitution de domain, et de bus n'est pas
  * effectuée si on stringifie directement dans la macro GenerateIvyBus */
Index: trunk/src/timer.h
===================================================================
--- trunk/src/timer.h	(revision 3649)
+++ trunk/src/timer.h	(working copy)
@@ -36,7 +36,7 @@
 
 
  //  implemetation of gettimeofday for windows
-#ifdef WIN32
+#if defined(WIN32) && !defined(__MINGW32__)
 #include "time.h"
 struct timezone 
 {
