--- my_config.h	2015-05-30 12:54:23.764910900 +0800
+++ my_config.h.new	2015-05-30 12:58:19.860414800 +0800
@@ -16,6 +16,12 @@
 #ifndef MY_CONFIG_H
 #define MY_CONFIG_H
 
+#if defined(__MINGW32__)
+#define _TIMESPEC_DEFINED
+#define _MODE_T_
+#define _SSIZE_T_DEFINED
+#endif
+
 /*
  * From configure.cmake, in order of appearance 
  */
@@ -195,7 +201,7 @@
 /* #undef HAVE_BUILTIN_UNREACHABLE */
 /* #undef HAVE_BUILTIN_EXPECT */
 /* #undef HAVE_BUILTIN_STPCPY */
-/* #undef HAVE_GCC_ATOMIC_BUILTINS */
+#define HAVE_GCC_ATOMIC_BUILTINS 1
 /* #undef HAVE_VALGRIND */
 
 /* IPV6 */
